.github/workflows/conda-cpp-build.yaml

```diff
@@ -96,16 +121,60 @@ jobs:
           repository: ${{ inputs.repo }}
           ref: ${{ inputs.sha }}
           fetch-depth: 0
+      - name: Checkout actions
+        uses: actions/checkout@v4
+        with:
+            repository: rapidsai/shared-actions
+            ref: ${{inputs.shared_actions_ref}}
+            path: ./shared-actions
       - name: Standardize repository information
         run: |
           echo "RAPIDS_REPOSITORY=${{ inputs.repo || github.repository }}" >> "${GITHUB_ENV}"
           echo "RAPIDS_SHA=$(git rev-parse HEAD)" >> "${GITHUB_ENV}"
           echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" >> "${GITHUB_ENV}"
           echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" >> "${GITHUB_ENV}"
+
+      # Temporary until https://github.com/rapidsai/ci-imgs/pull/196 is merged
+      - name: Install OpenTelemetry instrumentation
+        run: |
+          pip install opentelemetry-distro[otlp] \
+              opentelemetry-exporter-prometheus \
+              git+https://github.com/msarahan/opentelemetry-python-contrib.git@add-conda-build-instrumentation#subdirectory=instrumentation/opentelemetry-instrumentation-conda-build
+          opentelemetry-bootstrap -a install
+
+      - name: Setup proxy cache
+        uses: nv-gha-runners/setup-proxy-cache@main
```

Comment:
was this addition intended?

I think we should save this for a separate PR to avoid introducing too many changes in this PR.